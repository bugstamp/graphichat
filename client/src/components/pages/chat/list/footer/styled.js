import styled from 'styled-components';

import { getSpacing } from '../../../../../styles';

export const ChatListFooterStyled = styled.div`
  display: flex;
  justify-content: center;
  padding: ${getSpacing(2)} ${getSpacing(1)};

  && {
    button {
      width: 50%;
    }
  }
`;

export default null;
