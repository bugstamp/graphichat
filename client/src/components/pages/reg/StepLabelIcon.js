import React from 'react';
import PropTypes from 'prop-types';

import AccountCircleIcon from '@material-ui/icons/AccountCircleRounded';
import InfoIcon from '@material-ui/icons/InfoRounded';
import CheckCircleIcon from '@material-ui/icons/CheckCircleRounded';

const StepLabelIcon = ({ isCompleted, step, iconColor }) => (
  <Choose>
    <When condition={isCompleted}>
      <CheckCircleIcon color={iconColor} />
    </When>
    <When condition={step === 1}>
      <AccountCircleIcon color={iconColor} />
    </When>
    <Otherwise>
      <InfoIcon color={iconColor} />
    </Otherwise>
  </Choose>
);

StepLabelIcon.propTypes = {
  isCompleted: PropTypes.bool.isRequired,
  step: PropTypes.number.isRequired,
  iconColor: PropTypes.string.isRequired,
};

export default StepLabelIcon;
