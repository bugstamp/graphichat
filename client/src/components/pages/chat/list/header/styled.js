import styled from 'styled-components';
import { position } from 'polished';

import IconButton from '@material-ui/core/IconButton';

import { getSpacing } from '../../../../../styles';

export const ChatListHeaderStyled = styled.div`
  display: flex;
  flex-flow: column;
  padding: ${getSpacing(2)} ${getSpacing(1)};
`;

export const HeaderTitle = styled.div`
  position: relative;
  margin-bottom: ${getSpacing(1)};
`;

export const AddButton = styled(IconButton)`
  && {
    ${position('absolute', 0, 0, null, null)};
    padding: ${getSpacing(0.5)};
  }
`;
