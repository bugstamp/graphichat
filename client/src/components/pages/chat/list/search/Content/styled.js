import styled from 'styled-components';

import DialogContent from '@material-ui/core/DialogContent';

import { SearchBoxStyled } from '../../../../../common/SearchBox/styled';
import { getStyledProps, getSpacing } from '../../../../../../styles';

export const SearchDialogContentStyled = styled(DialogContent)`
  && {
    flex: 1 auto;
    display: flex;
    flex-flow: column;
    padding: ${getSpacing(2)} ${getSpacing(3)};
    padding-bottom: 0;

    ${SearchBoxStyled} {
      margin-bottom: ${getSpacing(2)}
    }

  ${(props) => {
    const breakpoints = getStyledProps('theme.breakpoints')(props);
    const spacing = getStyledProps('theme.spacing')(props);
    const smDown = breakpoints.down('sm');

    return `
      ${SearchBoxStyled} {
        margin-bottom: ${getSpacing(1)}
      }
      ${smDown} {
        padding: ${spacing(1)}px ${spacing(2)}px;
      }
    `;
  }}
  }
`;

export default null;
