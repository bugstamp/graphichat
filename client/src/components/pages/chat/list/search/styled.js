import styled from 'styled-components';

import { getStyledProps } from '../../../../../styles';

export const SearchDialogBodyStyled = styled.div`
  && {
    display: flex;
    flex-flow: column;
    background-color: #fff;

  ${(props) => {
    const breakpoints = getStyledProps('theme.breakpoints')(props);
    const borderRadius = getStyledProps('theme.shape.borderRadius')(props);
    const shadows = getStyledProps('theme.shadows[1]')(props);
    const mdUp = breakpoints.up('md');

    return `
      ${mdUp} {
        min-height: 50vmin;
        max-height: 90vmin;
        border-radius: ${borderRadius}px;
        box-shadow: ${shadows};
      }
    `;
  }}
  }
`;

export default null;
