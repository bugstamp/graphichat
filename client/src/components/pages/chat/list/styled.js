import styled from 'styled-components';

import Paper from '@material-ui/core/Paper';

import { getStyledProps } from '../../../../styles';

export const ChatListStyled = styled(Paper)`
  && {
    position: relative;
    height: 100%;
    display: flex;
    flex-flow: column;
    background-color: ${getStyledProps('theme.palette.background.default')};
    border-left: 1px solid ${getStyledProps('theme.palette.grey.200')};
    border-right: 1px solid ${getStyledProps('theme.palette.grey.200')};
  }
`;
