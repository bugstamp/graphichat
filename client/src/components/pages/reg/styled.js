import styled, { keyframes } from 'styled-components';
import { backgrounds } from 'polished';
import {
  fadeIn,
  fadeInRight,
  fadeInLeft,
  hinge,
} from 'react-animations';

import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';
import StepLabel from '@material-ui/core/StepLabel';

import { getStyledProps, getSpacing } from '../../../styles';
import bgImage from '../../../assets/images/reg-bg__1920_65.jpg';

const fadeInAnimation = keyframes`${fadeIn}`;
const fadeInRightAnimation = keyframes`${fadeInRight}`;
const fadeInLeftAnimation = keyframes`${fadeInLeft}`;
const hingeAnimation = keyframes`${hinge}`;

export const RegWrapper = styled(Grid)`
  && {
    flex: 1 auto;
    display: flex;
    flex-flow: row nowrap;
    position: relative;
    ${backgrounds(`url(${bgImage})`, 'no-repeat')}
    background-size: cover;
    background-position: center;
    overflow: hidden;

  ${(props) => {
    const breakpoints = getStyledProps('theme.breakpoints')(props);
    const smDown = breakpoints.down('sm');

    return `
      ${smDown} {
        flex-flow: column;
      }
    `;
  }}
  }
`;

export const RegFormWrapper = styled.div`
  flex: 1 40%;
  max-height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  opacity: 0;
  animation: 1.5s ${fadeInAnimation};
  animation-delay: 1s;
  animation-fill-mode: forwards;

  ${(props) => {
    const breakpoints = getStyledProps('theme.breakpoints')(props);
    const smDown = breakpoints.down('sm');

    return `
      ${smDown} {
        animation-play-state: paused;
        opacity: 1;
      }
    `;
  }}
`;

export const RegPresentationWrapper = styled.div`
  flex: 1 60%;
  display: flex;
  flex-flow: column;
  align-items: center;
  justify-content: center;

  ${(props) => {
    const breakpoints = getStyledProps('theme.breakpoints')(props);
    const smDown = breakpoints.down('sm');

    return `
      ${smDown} {
        display: none;
      }
    `;
  }}
`;

export const SubTitle = styled.div`
  color: #fff;
  text-align: center;
`;

export const SubTitleWord = styled(Typography)`
  && {
    display: inline;
    margin-right: ${getSpacing(1)};
    font-weight: bold;
    opacity: 0;
    animation: 1s ${({ order }) => (order === 'even' ? fadeInLeftAnimation : fadeInRightAnimation)};
    animation-delay: ${({ delay }) => `${delay}s`};
    animation-fill-mode: forwards;
  }
`;

export const Enjoy = styled.div`
  margin-top: ${getSpacing(3)};
  color: #fff;
  opacity: 0;
  animation: 2s ${hingeAnimation};
  animation-delay: 2s;
  animation-fill-mode: forwards;

  p {
    font-weight: bold;
  }
`;

export const FormWrapper = styled(Paper)`
  && {
    width: 100%;
    max-height: 100%;
    max-width: 375px;
    min-width: 320px;
    position: relative;
    margin: auto 0;
    padding: ${getSpacing(4)} ${getSpacing(3)};
    overflow-y: auto;

  ${(props) => {
    const breakpoints = getStyledProps('theme.breakpoints')(props);
    const spacing = getStyledProps('theme.spacing')(props);
    const smDown = breakpoints.down('sm');
    const xsDown = breakpoints.down('xs');

    return `
      ${smDown} {
        max-width: 100%;
        height: 100%;
        border-radius: 0;
        padding: ${spacing(4)}px ${spacing(20)}px;
      }
      ${xsDown} {
        padding: ${spacing(2)}px;
      }
    `;
  }}
  }
`;

export const FormHeader = styled(Typography)`
  && {
    width: 100%;
    position: relative;
  }
`;

export const FormInfo = styled(Typography)`
  && {
    text-align: center;
  }
`;

export const FormFooter = styled(Typography)`
  && {
    width: 100%;
    margin-top: ${getSpacing(2)};
  }
`;

export const StepLabelStyled = styled(StepLabel)`
  p {
    width: 100%;
  }
`;
