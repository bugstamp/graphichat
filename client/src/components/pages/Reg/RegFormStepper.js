import React from 'react';
import PropTypes from 'prop-types';
import { map } from 'lodash';

import Stepper from '@material-ui/core/Stepper';
import Step from '@material-ui/core/Step';

import StepLabelIcon from './StepLabelIcon';

import { StepLabelStyled as StepLabel } from './styled';

const RegFormStepper = ({ activeStep, steps }) => (
  <Stepper alternativeLabel activeStep={activeStep}>
    {
      map(steps, (label, index) => {
        const isActive = activeStep === index;
        const isCompleted = index < activeStep;
        const iconColor = (!isCompleted && !isActive)
          ? 'disabled'
          : 'primary';
        const step = index + 1;

        return (
          <Step
            key={index}
            active={isActive}
            completed={isCompleted}
          >
            <StepLabel
              icon={(<StepLabelIcon isCompleted={isCompleted} step={step} iconColor={iconColor} />)}
            >
              <span>{`Step ${step}.`}</span>
              <p>{label}</p>
            </StepLabel>
          </Step>
        );
      })
    }
  </Stepper>
);

RegFormStepper.propTypes = {
  activeStep: PropTypes.number.isRequired,
  steps: PropTypes.arrayOf(PropTypes.string).isRequired,
};

export default RegFormStepper;
