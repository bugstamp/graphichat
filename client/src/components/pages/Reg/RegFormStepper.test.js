import React from 'react';
import { shallow } from 'enzyme';

import Stepper from '@material-ui/core/Stepper';
import Step from '@material-ui/core/Step';

import { RegFormStepper, steps } from './RegFormStepper';
import StepLabelIcon from './StepLabelIcon';
import { StepLabelStyled as StepLabel } from './styled';

import { mountMockedProvider } from '../../../__mocks__/mockedProvider';

describe('RegFormStepper', () => {
  let activeStep = 0;
  const mountWrapper = () => mountMockedProvider(<RegFormStepper activeStep={activeStep} />);

  test('should match to snapshot', () => {
    expect(shallow(<RegFormStepper activeStep={0} />)).toMatchSnapshot();
  });
  test('should mount without errors', () => {
    const wrapper = mountWrapper();

    expect(wrapper.find(RegFormStepper)).toHaveLength(1);
  });
  test('should have Stepper with correct props', () => {
    const wrapper = mountWrapper();

    expect(wrapper.find(Stepper)).toHaveLength(1);
    expect(wrapper.find(Stepper).props()).toMatchObject({
      alternativeLabel: true,
      activeStep,
    });
  });
  describe('Step', () => {
    test('paragraph text should be matched to steps array', () => {
      const wrapper = mountWrapper();
      const paragraphs = wrapper.find(Step).find('p');

      expect(wrapper.find(Step)).toHaveLength(steps.length);

      paragraphs.forEach((p, index) => {
        expect(p.text()).toEqual(steps[index]);
      });
    });
    test('active Step props', () => {
      const wrapper = mountWrapper();
      const ActiveStepWrapper = wrapper.find(Step).at(activeStep);

      expect(ActiveStepWrapper).toHaveLength(1);
      expect(ActiveStepWrapper.props()).toMatchObject({
        active: true,
        completed: false,
      });
    });
    test('inactive Step props', () => {
      const wrapper = mountWrapper();
      const InactiveStepWrapper = wrapper.find(Step).at(activeStep + 1);

      expect(InactiveStepWrapper).toHaveLength(1);
      expect(InactiveStepWrapper.props()).toMatchObject({
        active: false,
        completed: false,
      });
    });
    test('active StepLabel props', () => {
      const wrapper = mountWrapper();
      const ActiveStepWrapper = wrapper.find(Step).at(activeStep);
      const ActiveStepLabelWrapper = ActiveStepWrapper.find(StepLabel);

      expect(ActiveStepLabelWrapper).toHaveLength(1);
      expect(ActiveStepLabelWrapper.find(StepLabelIcon).props()).toMatchObject({
        isCompleted: false,
        step: activeStep + 1,
        iconColor: 'primary',
      });
      expect(ActiveStepLabelWrapper.find('span.step').text()).toMatch(`Step ${activeStep + 1}.`);
      expect(ActiveStepLabelWrapper.find('p').text()).toEqual(steps[activeStep]);
    });
    test('completed prop should be true in the passed steps', () => {
      activeStep = 1;
      const wrapper = mountWrapper();
      const PassedStepWrapper = wrapper.find(Step).at(0);

      expect(PassedStepWrapper).toHaveLength(1);
      expect(PassedStepWrapper.props()).toMatchObject({
        active: false,
        completed: true,
      });
    });
  });
});
