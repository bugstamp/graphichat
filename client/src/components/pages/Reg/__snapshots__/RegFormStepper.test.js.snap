// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegFormStepper should match to snapshot 1`] = `
<WithStyles(ForwardRef(Stepper))
  activeStep={0}
  alternativeLabel={true}
>
  <WithStyles(ForwardRef(Step))
    active={true}
    completed={false}
    key="0"
  >
    <styled__StepLabelStyled
      icon={
        <StepLabelIcon
          iconColor="primary"
          isCompleted={false}
          step={1}
        />
      }
    >
      <span
        className="step"
      >
        Step 1.
      </span>
      <p>
        Create your account
      </p>
    </styled__StepLabelStyled>
  </WithStyles(ForwardRef(Step))>
  <WithStyles(ForwardRef(Step))
    active={false}
    completed={false}
    key="1"
  >
    <styled__StepLabelStyled
      icon={
        <StepLabelIcon
          iconColor="disabled"
          isCompleted={false}
          step={2}
        />
      }
    >
      <span
        className="step"
      >
        Step 2.
      </span>
      <p>
        Tell us about yourself
      </p>
    </styled__StepLabelStyled>
  </WithStyles(ForwardRef(Step))>
</WithStyles(ForwardRef(Stepper))>
`;
